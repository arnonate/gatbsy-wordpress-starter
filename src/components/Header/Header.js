import { Link, graphql, useStaticQuery } from "gatsby"
import React from "react"
import Img from "gatsby-image"

const Header = () => {
  const data = useStaticQuery(graphql`
    query {
      logo: file(relativePath: { eq: "logo.png" }) {
        childImageSharp {
          fluid(maxWidth: 400) {
            ...GatsbyImageSharpFluid_withWebp_noBase64
          }
        }
      }
    }
  `)

  return (
    <>
      <Link to="/">
        <Img
          fluid={data.logo.childImageSharp.fluid}
          alt="Wanna"
          loading="eager"
        />
      </Link>
    </>
  )
}

export default Header
