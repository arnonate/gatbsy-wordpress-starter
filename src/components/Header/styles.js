import styled from "styled-components"

import { Tokens } from "../Theme/Tokens"

export const Header = styled.header`
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  height: auto;
  background-color: ${Tokens.color.white};
  overflow: visible;
  padding: ${Tokens.rhythm} 0;
  font-weight: ${Tokens.font.weight.bold};
  z-index: ${Tokens.z.header};
  box-shadow: 2px 0px 4px rgba(0, 0, 0, 0.3);
`

export const HeaderBand = styled.div`
  position: relative;
  overflow: visible;
  z-index: ${Tokens.z.band};
`

export const HeaderBackdrop = styled.div`
  position: absolute;
  top: 100%;
  left: 50%;
  width: 240px;
  transform: translateX(-50%);
`

export const HeaderLogo = styled.div`
  position: absolute;
  top: 0;
  left: 50%;
  width: 110px;
  transform: translateX(-50%);
`

export const HeaderActions = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  max-width: 300px;
  display: none;
  margin: 0.8rem 1rem;
  text-align: center;
  z-index: ${Tokens.z.language};

  @media (min-width: ${Tokens.screen.medium}) {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 14%;
  }

  @media (min-width: ${Tokens.screen.large}) {
    padding-left: 2%;
    border-left: 2px solid ${Tokens.color.light};
  }

  span {
    color: ${Tokens.color.white};
    font-size: ${Tokens.font.size.micro};
  }
`

export const LanguageBubble = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  width: 100%;
  height: auto;
  transform: translate(-50%, -50%);
  z-index: ${Tokens.z.behind};
`

export const LanguageSelector = styled.div`
  position: relative;
  width: 38px;
  height: 28px;
  line-height: 1.3;
  cursor: pointer;
  text-transform: uppercase;
`

export const LinkGroups = styled.div`
  display: none;

  @media (min-width: ${Tokens.screen.medium}) {
    display: flex;
    justify-content: space-between;
  }
`

export const LinkGroup = styled.div`
  display: flex;
  justify-content: space-around;
  width: 40%;
  overflow: visible;

  & > * {
    font-size: ${Tokens.font.size.subBase};
    font-weight: ${Tokens.font.weight.bold};
    margin: 0 ${Tokens.font.size.small};
    text-decoration: none;
    color: ${Tokens.color.dark};
    white-space: nowrap;
  }

  .gold {
    color: ${Tokens.color.gold};
  }
`

export const MobileMenu = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 85px;
  line-height: 0.5;
  color: ${Tokens.color.default};
  height: 22px;
  font-size: ${Tokens.font.size.base};
  padding-left: 1rem;
  cursor: pointer;
  user-select: none;

  @media (min-width: ${Tokens.screen.medium}) {
    display: none;
  }
`

export const Hamburger = styled.div`
  display: flex;
  flex-direction: column;
  height: 14px;

  span {
    line-height: 0;
  }

  span + span {
    margin-top: 6px;
  }
`
