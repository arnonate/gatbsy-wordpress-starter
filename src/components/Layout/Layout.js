import React, { useState, useEffect } from "react"
import { ParallaxProvider } from "react-scroll-parallax"
import { navigate } from "gatsby"
import { createGlobalStyle } from "styled-components"

import { Context } from "../Context"
import "../Theme/reset.css"
import {
  Footer,
  Header,
  AppModal,
  LanguageModal,
  MenuModal,
} from "../../components"

const Layout = ({ children }) => {
  const defaultLang =
    (typeof window !== "undefined" &&
      window.localStorage.getItem("wannaLang")) ||
    (typeof window !== "undefined" && window.navigator.language.split("-")[0])

  const [language, toggleLanguage] = useState(defaultLang ? defaultLang : "en")
  const [appModalIsVisible, toggleAppModalIsVisible] = useState(false)
  const [languageModalIsVisible, toggleLanguageModalIsVisible] = useState(false)
  const [menuModalIsVisible, toggleMenuModalIsVisible] = useState(false)

  useEffect(() => {
    const langList = ["pt"]

    const path =
      typeof window !== "undefined" &&
      window.location.pathname
        .split("/")
        .filter(item => item.length && !langList.includes(item))

    const hash =
      typeof window !== "undefined" && window.location.hash
        ? window.location.hash
        : ""

    if (language !== "en" && !path.includes(language)) {
      const uri = `/${language ? language + "/" : ""}${
        path.length > 0 ? path.join("/") + "/" : ""
      }${hash}`
      navigate(uri)
    }
    if (language === "en") {
      const uri = `/${path.length > 0 ? path.join("/") + "/" : ""}${hash}`
      navigate(uri)
    }
  }, [language])

  const context = {
    language,
    toggleAppModalIsVisible,
    appModalIsVisible,
    toggleLanguage,
    languageModalIsVisible,
    toggleLanguageModalIsVisible,
    menuModalIsVisible,
    toggleMenuModalIsVisible,
  }

  const updateLanguage = lang => {
    toggleLanguage(lang)
    toggleLanguageModalIsVisible(false)
    toggleMenuModalIsVisible(false)

    if (typeof window !== "undefined") {
      window.localStorage.setItem("wannaLang", lang)
    }
  }

  return (
    <Context.Provider value={context}>
      <GlobalStyle
        modalIsVisible={languageModalIsVisible || menuModalIsVisible}
      />

      <Header />

      <ParallaxProvider>
        <main>{children}</main>
      </ParallaxProvider>

      <Footer />

      <AppModal isVisible={appModalIsVisible} />

      <LanguageModal
        isVisible={languageModalIsVisible}
        onUpdate={updateLanguage}
        onClose={toggleLanguageModalIsVisible}
        currentLanguage={language}
      />

      <MenuModal
        isVisible={menuModalIsVisible}
        onLanguageUpdate={updateLanguage}
      />
    </Context.Provider>
  )
}

const GlobalStyle = createGlobalStyle`
  body {
    overflow: ${props => (props.modalIsVisible ? "hidden" : "auto")};
  }
`

export default Layout
