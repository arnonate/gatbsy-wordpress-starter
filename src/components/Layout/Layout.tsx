import React from "react"
import { useStaticQuery, graphql } from "gatsby"
import { FluidObject } from "gatsby-image"

import { Footer, Header } from "../../components"
import "../../styles/reset.css"
import "../../styles/global.css"

export type DataProps = {
  logo: {
    childImageSharp: {
      fluid: FluidObject
    }
  }
  site: {
    siteMetadata: {
      title: string
    }
  }
}

type ComponentProps = {
  children: JSX.Element
  data: Readonly<DataProps>
}

export const LayoutComponent = ({
  data,
  children,
}: Readonly<ComponentProps>): JSX.Element => (
  <>
    <Header data={data} />
    <main>{children}</main>
    <Footer data={data} />
  </>
)

const Layout = ({ children }: ComponentProps) => {
  const data = useStaticQuery<DataProps>(graphql`
    query {
      logo: file(relativePath: { eq: "logo.png" }) {
        childImageSharp {
          fluid(maxWidth: 400) {
            ...GatsbyImageSharpFluid_withWebp_noBase64
          }
        }
      }
      site {
        siteMetadata {
          title
        }
      }
    }
  `)

  return <LayoutComponent data={data}>{children}</LayoutComponent>
}

export default Layout
