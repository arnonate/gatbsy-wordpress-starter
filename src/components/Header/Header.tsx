import { Link } from "gatsby"
import React from "react"
import Img from "gatsby-image"

import { DataProps as LayoutDataProps } from "../Layout/Layout"

type ComponentProps = {
  data?: LayoutDataProps
}

const Header = ({ data }: Readonly<ComponentProps>): JSX.Element => (
  <header>
    <div className="styled-header">
      <div className="girdle flex flex-column flex-center">
        <div className="logo">
          <Link to="/" title="Go to Home Page">
            {data && (
              <Img
                alt={data.site.siteMetadata.title}
                fluid={data.logo.childImageSharp.fluid}
              />
            )}
          </Link>
        </div>

        <nav aria-label="Primary">
          <ul className="nav flex flex-between">
            <li>
              <Link to="/" activeClassName="active">
                HOME
              </Link>
            </li>
            <li>
              <Link to="/about/" activeClassName="active">
                ABOUT
              </Link>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
)

export default Header
