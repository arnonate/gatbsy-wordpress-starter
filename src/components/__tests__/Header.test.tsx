import React from "react"
import { render } from "@testing-library/react"

import Header from "../Header/Header"
import { DataProps as LayoutDataProps } from "../Layout/Layout"

const data: LayoutDataProps = {
  logo: {
    childImageSharp: {
      fluid: {
        aspectRatio: 1,
        src: "img src",
        srcSet: "img srcSet",
        sizes: "img sizes",
      },
    },
  },
  site: {
    siteMetadata: {
      title: "Headless",
    },
  },
}

describe("Header", () => {
  it("renders correctly", () => {
    const component = <Header data={data} />

    const { container } = render(component)

    expect(container.firstElementChild).toMatchSnapshot()
  })
})
