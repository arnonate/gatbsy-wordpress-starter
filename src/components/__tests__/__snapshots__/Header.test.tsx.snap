// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <div
    class="styled-header"
  >
    <div
      class="girdle flex flex-column flex-center"
    >
      <div
        class="logo"
      >
        <a
          href="/"
          title="Go to Home Page"
        >
          <div
            class="gastby-image gatsby-image-wrapper"
            style="position: relative; overflow: hidden;"
          >
            <div
              aria-hidden="true"
              style="width: 100%; padding-bottom: 100%;"
            />
            <picture>
              <source
                sizes="img sizes"
                srcset="img srcSet"
              />
              <img
                alt="Headless"
                loading="lazy"
                sizes="img sizes"
                src="img src"
                srcset="img srcSet"
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
              />
            </picture>
            <noscript>
              &lt;picture&gt;&lt;source srcset="img srcSet" sizes="img sizes" /&gt;&lt;img loading="lazy" sizes="img sizes" srcset="img srcSet" src="img src" alt="Headless" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
            </noscript>
          </div>
        </a>
      </div>
      <nav
        aria-label="Primary"
      >
        <ul
          class="nav flex flex-between"
        >
          <li>
            <a
              href="/"
            >
              HOME
            </a>
          </li>
          <li>
            <a
              href="/about/"
            >
              ABOUT
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
